src/Elastic.Apm.Azure.Functions/AzureFunctionsDiagnosticSubscriber.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using Elastic.Apm.DiagnosticSource;
+using Elastic.Apm.Logging;
+
+namespace Elastic.Apm.Azure.Functions
+{
+	public class AzureFunctionsDiagnosticSubscriber : IDiagnosticsSubscriber
```

Comment:
Why can't we reuse our normal AspNetCore subscriber here? https://github.com/elastic/apm-agent-dotnet/blob/31736e4a8ec3ff81cdac514f44f6b745488e5b9d/src/Elastic.Apm.AspNetCore/DiagnosticListener/AspNetCoreDiagnosticListener.cs#L19

@gregkalapos

Reply:
Very good point - I think we could.

We set `Transaction.FaaS` here, that needs to be done, but other than that I think this is the same as the ASP.NET Core one.